server/testdriver/core/src/main/java/org/infinispan/server/test/core/rollingupgrade/RollingUpgradeConfigurationBuilder.java

```diff
@@ -0,0 +1,107 @@
+package org.infinispan.server.test.core.rollingupgrade;
+
+import java.net.SocketAddress;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.commons.configuration.StringConfiguration;
+
+public class RollingUpgradeConfigurationBuilder {
+   private final String fromVersion;
+   private final String toVersion;
+
+   private int nodeCount = 3;
+   private boolean xSite;
+   private String jgroupsProtocol = "tcp";
+   private int serverCheckTimeSecs = 30;
+   private boolean useSharedDataMount = true;
+   private BiConsumer<Throwable, RollingUpgradeHandler> exceptionHandler = (t, uh) -> {
+      t.printStackTrace();
+   };
+
+   private Consumer<String> logConsumer = System.out::println;
+
+   private Consumer<RollingUpgradeHandler> initialHandler = uh -> {
+      RemoteCacheManager rcm = uh.getRemoteCacheManager();
+      RemoteCache<String, String> cache = rcm.administration().createCache("rolling-upgrade",
+            new StringConfiguration("<replicated-cache></replicated-cache>"));
+      cache.put("foo", "bar");
+   };
```

Comment:
Is this just simplifying for our use? The cache name + configuration should be required options in the builder, and we do this automatically.